src/main/java/org/generationcp/middleware/dao/dms/InstanceDao.java

```diff
@@ -0,0 +1,482 @@
+package org.generationcp.middleware.dao.dms;
+
+import org.apache.commons.lang3.math.NumberUtils;
+import org.generationcp.middleware.dao.GenericDAO;
+import org.generationcp.middleware.domain.dms.LocationDto;
+import org.generationcp.middleware.domain.dms.StudyReference;
+import org.generationcp.middleware.domain.dms.TrialEnvironment;
+import org.generationcp.middleware.domain.dms.TrialEnvironmentProperty;
+import org.generationcp.middleware.domain.dms.TrialEnvironments;
+import org.generationcp.middleware.domain.oms.TermId;
+import org.generationcp.middleware.enumeration.DatasetTypeEnum;
+import org.generationcp.middleware.exceptions.MiddlewareQueryException;
+import org.generationcp.middleware.pojos.dms.ExperimentModel;
+import org.hibernate.Criteria;
+import org.hibernate.HibernateException;
+import org.hibernate.Query;
+import org.hibernate.SQLQuery;
+import org.hibernate.criterion.DetachedCriteria;
+import org.hibernate.criterion.Projections;
+import org.hibernate.criterion.Property;
+import org.hibernate.criterion.Restrictions;
+import org.hibernate.criterion.Subqueries;
+import org.hibernate.type.IntegerType;
+import org.hibernate.type.StringType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.CollectionUtils;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class InstanceDao extends GenericDAO<ExperimentModel, Integer> {
```

Comment:
Remove unused methods (these are copied from the Geolocation Refactoring branch but not actually used now):
```
getEnvironmentIds
getAllTrialEnvironments
countAllTrialEnvironments
getPropertiesForTrialEnvironments
getTrialEnvironmentDetails
getEnvironmentsForTraits
getNextInstanceNumber
getInstanceMetadata
getEnvironmentIdByStudyNameAndInstanceNumberAndProgramUUID
getExperimentIdEnvironmentMap
```

Reply:
Done